### DISCUSSION TITLE
Adding zod as a schema source

### DISCUSSION THREAD

**Discussion by tevonsb** [microsoft/TypeChat#113] [Ideas]
Hi All, 

I love the thinking behind TypeChat and we will likely incorporate it into our product.

We use Zod extensively to define complex objects along with descriptions of the various fields as well as custom validations.

I think it fits very well with the concept of typechat and would love to see it integrated, since zod extends the typescript definition to support more complex usecases and adds descriptions and documentation directly into the object.

It seems to fit well both with the type definitions (as zod can produce type definitions) as well as the validation section. 

One additional benefit of Zod is that it does not have to be defined statically, but can be specified dynamically, which would significantly increase the flexibility of typechat by allowing clients to dynamically change schema based on the history of the conversation, the user, or any other dynamic data.

We'd love to be helpful if this is something that is in-scope.

**Comment by ahejlsberg** [comment]
@tevonsb Agreed about Zod support. Now implemented in https://github.com/microsoft/TypeChat/pull/147.

**Reply by tevonsb** [reply]
Thanks @ahejlsberg excited to check it out.

**Comment by DanielRosenwasser** [comment]
You can now use this in TypeChat 0.1.0: https://microsoft.github.io/TypeChat/blog/announcing-typechat-0-1-0/
